Starforge/Editor/Camera.cs

```diff
@@ -91,24 +113,27 @@ public class Camera {
             VisibleArea = new Rectangle((int)min.X, (int)min.Y, (int)(max.X - min.X), (int)(max.Y - min.Y));
 
             // Invoke position change event
-            PositionChange?.Invoke();
+            OnPositionChange?.Invoke();
         }
 
-        public void UpdateViewport() {
-            Viewport = Engine.Instance.GraphicsDevice.Viewport;
-            Update();
-        }
+        /// <summary>
```

Comment:
What do you mean by this exactly?

Reply:
Instead of just being able to zoom by *2 or /2, make the zoom multiplier an argument. You could for example make the zoom "speed" make a setting when doing that. Also the code is kinda redundant except for that one /2 or *2 line

Reply:
As for the first part - zoom is always halved or doubled because going to other values between results in individual pixels of the map not scaling properly and turning into rectangles instead of perfect squares, which looks bad.

The code could probably be improved as well, yes, but I don't remember getting it to center on the mouse cursor properly without.. what is already there. The Update() calls at the end can be removed, though.